python/tvm/meta_schedule/integration.py

```diff
@@ -230,45 +213,32 @@ def extract_task_from_relay(
         The tasks extracted from this network
     """
 
-    @contextmanager
-    def _autotvm_silencer():
-        from tvm import autotvm  # pylint: disable=import-outside-toplevel
-
-        silent = autotvm.GLOBAL_SCOPE.silent
-        autotvm.GLOBAL_SCOPE.silent = True
-        try:
-            yield
-        finally:
-            autotvm.GLOBAL_SCOPE.silent = silent
+    extract_task_func = get_global_func("relay.backend.MetaScheduleExtractTask")
+    assert extract_task_func
 
-    def _thread_run(func: Callable[[], None]) -> None:
-        import threading  # pylint: disable=import-outside-toplevel
+    target = Target(target) if isinstance(target, str) else target
 
-        thread = threading.Thread(target=func)
-        thread.start()
-        thread.join()
+    relay_params = {}
+    for name, param in params.items():
+        if isinstance(param, np.ndarray):
+            param = nd.array(param)
+        relay_params[name] = const(param)
```

Comment:
Would you mind elaborating why we need to convert NDArray to relay::Constant? Thanks a lot!

Reply:
Initially I was strictly following what `VMCompiler.lower()` based task extraction does. This was copied from https://github.com/apache/tvm/blob/137def8bf211ccad165349d9f4ba7d20f3a7164e/python/tvm/relay/backend/vm.py#L109

But yeah, this is not necessary at all. I changed the interface of `ExtractTask` to take in `NDArray` directly, rather than `Constant`.

Reply:
Thanks for your patient explanation! It makes total sense to me now! Sorry I wasn't even aware :-)